include/exadg/structure/material/library/incompressible_neo_hookean.cpp

```diff
@@ -0,0 +1,166 @@
+                          this,
+                          dummy,
+                          dummy);
+  }
+}
+
+template<int dim, typename Number>
+void
+IncompressibleNeoHookean<dim, Number>::cell_loop_set_coefficients(
+  dealii::MatrixFree<dim, Number> const & matrix_free,
+  VectorType &,
+  VectorType const &,
+  Range const & cell_range) const
+{
+  IntegratorCell integrator(matrix_free, dof_index, quad_index);
+
+  // loop over all cells
+  for(unsigned int cell = cell_range.first; cell < cell_range.second; ++cell)
+  {
+    integrator.reinit(cell);
+
+    // loop over all quadrature points
+    for(unsigned int q = 0; q < integrator.n_q_points; ++q)
+    {
+      dealii::VectorizedArray<Number> shear_modulus_vec =
+        FunctionEvaluator<0, dim, Number>::value(*(data.shear_modulus_function),
+                                                 integrator.quadrature_point(q),
+                                                 0.0 /*time*/);
+
+      // set the coefficients
+      shear_modulus_coefficients.set_coefficient_cell(cell, q, shear_modulus_vec);
+    }
+  }
+}
+
+template<int dim, typename Number>
+dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
+IncompressibleNeoHookean<dim, Number>::second_piola_kirchhoff_stress(
+  tensor const &     gradient_displacement,
+  unsigned int const cell,
+  unsigned int const q) const
+{
+  tensor S;
+
+  if(shear_modulus_is_variable)
+  {
+    shear_modulus_stored = shear_modulus_coefficients.get_coefficient_cell(cell, q);
+  }
+
+  tensor F     = get_F<dim, Number>(gradient_displacement);
+  scalar J     = determinant(F);
+  tensor C     = transpose(F) * F;
+  tensor C_inv = invert(C);
+
+  // S_vol, i.e., penalty term enforcing J = 1.
+  S = (data.bulk_modulus * 0.5 * (J * J - 1.0)) * C_inv;
+
+  // S_iso, isochoric term.
+  S += (shear_modulus_stored * pow(J, static_cast<Number>(-2.0 * one_third))) *
+       add_identity<dim, Number>((-one_third * trace(C)) * C_inv);
+
+  return S;
+}
+
+template<int dim, typename Number>
+dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
+IncompressibleNeoHookean<dim, Number>::second_piola_kirchhoff_stress_displacement_derivative(
+  tensor const &     gradient_increment,
+  tensor const &     deformation_gradient,
+  unsigned int const cell,
+  unsigned int const q) const
+{
+  tensor Dd_S;
+
+  if(shear_modulus_is_variable)
+  {
+    shear_modulus_stored = shear_modulus_coefficients.get_coefficient_cell(cell, q);
+  }
+
+  scalar J   = determinant(deformation_gradient);
+  tensor C   = transpose(deformation_gradient) * deformation_gradient;
+  scalar I_1 = trace(C);
+
+  tensor F_inv = invert(deformation_gradient);
+  tensor C_inv = invert(C);
+
+  scalar one_over_J_times_Dd_J = trace(F_inv * gradient_increment);
+  tensor Dd_F_inv              = -F_inv * (gradient_increment * F_inv);
+  tensor Dd_C_inv              = Dd_F_inv * transpose(F_inv) + F_inv * transpose(Dd_F_inv);
```

Comment:
use `tensor Dd_F_inv_times_transpose_F_inv = -F_inv * (gradient_increment * F_inv) * transpose(F_inv);`
and then `tensor Dd_C_inv = Dd_F_inv_times_transpose_F_inv + transpose(Dd_F_inv_times_transpose_F_inv);`